crates/sui-types/src/passkey_authenticator.rs

```diff
@@ -0,0 +1,250 @@
+        let serializable = RawPasskeyAuthenticator::deserialize(deserializer)?;
+        serializable
+            .try_into()
+            .map_err(|e: SuiError| Error::custom(e.to_string()))
+    }
+}
+
+impl PasskeyAuthenticator {
+    pub fn new_for_testing(
+        authenticator_data: Vec<u8>,
+        client_data_json: Vec<u8>,
+        user_signature: Signature,
+    ) -> Result<Self, SuiError> {
+        let raw = RawPasskeyAuthenticator {
+            authenticator_data,
+            client_data_json,
+            user_signature,
+        };
+        raw.try_into()
+    }
+
+    pub fn get_pk(&self) -> SuiResult<PublicKey> {
+        PublicKey::try_from_bytes(
+            SignatureScheme::PasskeyAuthenticator,
+            self.user_signature.public_key_bytes(),
+        )
+        .map_err(|_| SuiError::InvalidAuthenticator)
+    }
+}
+
+/// Necessary trait for [struct SenderSignedData].
+impl PartialEq for PasskeyAuthenticator {
+    fn eq(&self, other: &Self) -> bool {
+        self.as_ref() == other.as_ref()
+    }
+}
+
+/// Necessary trait for [struct SenderSignedData].
+impl Eq for PasskeyAuthenticator {}
+
+/// Necessary trait for [struct SenderSignedData].
+impl Hash for PasskeyAuthenticator {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.as_ref().hash(state);
+    }
+}
+
+impl AuthenticatorTrait for PasskeyAuthenticator {
+    fn verify_user_authenticator_epoch(
+        &self,
+        _epoch: EpochId,
+        _max_epoch_upper_bound_delta: Option<u64>,
+    ) -> SuiResult {
+        Ok(())
+    }
+
+    /// Verify an intent message of a transaction with an passkey authenticator.
+    fn verify_claims<T>(
+        &self,
+        intent_msg: &IntentMessage<T>,
+        author: SuiAddress,
+        _aux_verify_data: &VerifyParams,
+        _zklogin_inputs_cache: Arc<VerifiedDigestCache<ZKLoginInputsDigest>>,
+    ) -> SuiResult
+    where
+        T: Serialize,
+    {
+        // check parsed_challenge == intent || blake2b_hash(tx_data)
+        let digest = to_signing_digest(intent_msg);
+        if self.parsed_challenge != digest {
+            return Err(SuiError::InvalidSignature {
+                error: "Invalid challenge".to_string(),
+            });
+        };
+
+        // construct msg = authenticator_data || sha256(client_data_json)
+        let mut message = self.authenticator_data.clone();
+        let client_data_hash = Sha256::digest(self.client_data_json.as_slice()).digest;
+        message.extend_from_slice(&client_data_hash);
+
+        if author != SuiAddress::from(&self.get_pk()?) {
+            return Err(SuiError::InvalidSignature {
+                error: "Invalid author".to_string(),
+            });
+        };
+
+        let pk = self.pk.get().ok_or(SuiError::InvalidSignature {
+            error: "Missing pk".to_string(),
+        })?;
+        let signature = self.signature.get().ok_or(SuiError::InvalidSignature {
+            error: "Missing signature".to_string(),
+        })?;
```

Comment:
should these two just panic if its not there? they should be initialized as deserialization.

Reply:
the passkey authenticator is required to inherit jsonschema trait (from TransactionData) that requires a default impl for each field. the alternative is to impl default for Secp256r1PublicKey and Secp256r1Signature that after discussing with @benr-ml may not be appropriate. any other alternatives here you see?

Reply:
>the passkey authenticator is required to inherit jsonschema trait

you can hand impl the JsonSchema trait to make this easier

Reply:
realized that i dont need to have a default for jsonschema, removed once cell and everything just worked